tests/run-make/issue-11908/rmake.rs

```diff
@@ -0,0 +1,35 @@
+// A path which contains the same rlib or dylib in two locations
+// should not cause an assertion panic in the compiler.
+// This test tries to replicate the linked issue and checks
+// if the bugged error makes a resurgence.
+
+// Note that this relies on `liburl` to be in the path somewhere else.
+// Our aux-built libraries will collide with liburl (they have
+// the same version listed)
+
+// See https://github.com/rust-lang/rust/issues/11908
+
+//@ ignore-cross-compile
+
+use run_make_support::{dynamic_lib, rustc, tmp_dir};
+use std::fs;
+
+fn main() {
+    let mut tmp_dir_other = tmp_dir();
+    tmp_dir_other.push("other");
+
+    fs::create_dir(tmp_dir_other.clone());
+    rustc().input("foo.rs").crate_type("dylib").codegen_option("prefer-dynamic").run();
```

Comment:
Yeah, instead of `.codegen_option("prefer-dynamic")`, I think it's better if it's `.prefer_dynamic()`.